ADR/0022-secret-mgmt-for-user-workloads.md

```diff
@@ -26,65 +29,142 @@ Accepted
+
 ```yaml
 apiVersion: appstudio.redhat.com/v1beta1
 kind: RemoteSecret
 metadata:
-  name: jdbc-connection-parameter-prod
+    namespace: jdoe-tenant
 spec:
-  secret:
-    environment: prod
-    name: jdbc-connection-parameter
- ...
+    target:
+status:
+  TBD
 ```
 
-#### Example: If the destination is `Namespace` and secret has to be linked to SA
+#### Example:If multiple working Secrets should be created along with RemoteSecret.
 ```yaml
-apiVersion: v1
-kind: Secret
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
 metadata:
-  name: jdbc-connection-parameter
-  annotations:
-      spi.appstudio.redhat.com/secret-linkedto-serviceAccount-name: sa-star
-  labels:
-    spi.appstudio.redhat.com/upload-secret: secret
-    spi.appstudio.redhat.com/target-namespace: ns-9459039345
-type: Opaque
-data:
- ...
+    name: test-remote-secret-secret
+spec:
+    target:
+        - namespace: jdoe-test-ns
+          secretName: test-remote-secret-secret-5bmq9
+        - namespace: jdoe-prod-ns
+          secretName: test-remote-secret-secret-5bmq9
+status:
+    TBD
+```
+
+
+#### Example: If RemoteSecret has to be created with multiple target namespaces including BYC use-case
+```yaml
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
+metadata:
+    name: test-remote-secret-secret
+spec:
+    target:
+        - namespace: jdoe-test-ns
+          secretName: test-remote-secret-secret-5bmq9
+        - namespace: jdoe-prod-ns
+          secretName: test-remote-secret-secret-5bmq9
+          apiURL: https://somedomain.copm:443
+          clusterCredentialsSecret: team-a--prod-dtc--secret
+status:
+    TBD
 ```
+
+#### Example: If RemoteSecret has to be created with target namespace and Environment
+```yaml
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
+metadata:
+    name: test-remote-secret-secret
+    labels:
+        appstudio.redhat.com/remotesecret-target-environment: prod
+        appstudio.redhat.com/remotesecret-target-component: m-service
+        appstudio.redhat.com/remotesecret-target-application: coffee-shop
+spec:
+    target:
+        - namespace: jdoe-tenant
+          secretName: test-remote-secret-secret-5bmq9
+status:
+    TBD
+```
+
+#### Example: If RemoteSecret has to be created all Environments of certain component and application
+```yaml
+apiVersion: appstudio.redhat.com/v1beta1
+kind: RemoteSecret
+metadata:
+    name: test-remote-secret-secret
+    labels:
+        appstudio.redhat.com/remotesecret-target-component: m-service
+        appstudio.redhat.com/remotesecret-target-application: coffee-shop
+spec:
+    target:
+        - namespace: jdoe-tenant
+          secretName: test-remote-secret-secret-5bmq9
+status:
+    TBD
+```
+> **Note :** The component that will keep the connection between Environments and RemoteSecret's target should track all environments including the one that can be created after RemoteSecret creation.
```

Comment:
As far as I looked this is the first mention of some kind of external component that should be keeping the environment and the remote secret target(s) in sync. You mention this component and the reason for it to be TBD at the very end of this chapter, but IMHO it should be mentioned before the examples so that the readers can anticipate it.

Reply:
@metlos Agreed, we need the status which would reflect some state of working Secret. For start we can just reflect the fact that it is created, then we can check its healthy and maybe later even refresh its data.
And we can evolve the specification with it.